Bottom: 2c8b7627e2dcc1c904a9dfe50e66d0c750af8f06
Top:    573cce3a5b103daa57fa1ed3b30137e72123f0ca
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-01-09 12:35:17 +0800

polling

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index 5dc3465..6d004c6 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -689,14 +689,19 @@ static void handle_rx(struct vhost_net *net)
 		}
 		/* OK, now we need to know about added descriptors. */
 		if (!headcount) {
+			if (unlikely(!vhost_vq_avail_empty(&net->dev, vq)))
+				continue;
+#if 0
 			if (unlikely(vhost_enable_notify(&net->dev, vq))) {
 				/* They have slipped one in as we were
 				 * doing that: check again. */
 				vhost_disable_notify(&net->dev, vq);
 				continue;
 			}
+#endif
 			/* Nothing new?  Wait for eventfd to tell us
 			 * they refilled. */
+			vhost_poll_queue(&vq->poll);
 			goto out;
 		}
 		/* We don't need to be notified again. */
